fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreTest.java

```diff
@@ -3635,4 +3635,69 @@ public IndexState needRebuildIndex(Index index, long recordCount, boolean indexO
             assertTrue(recordStore.getRecordStoreState().getStoreHeader().getOmitUnsplitRecordSuffix());
         }
     }
+
+    @Test
+    public void testIndexStateFormatVersionUpgrade() {
+        final RecordMetaDataBuilder builder = RecordMetaData.newBuilder().setRecords(TestNoIndexesProto.getDescriptor());
+        FDBRecordStore.Builder storeBuilder = FDBRecordStore.newBuilder().setMetaDataProvider(builder).setKeySpacePath(path);
+        try (FDBRecordContext context = openContext()) {
+            recordStore = storeBuilder.setFormatVersion(FDBRecordStore.SAVE_INDEX_STATE_WITH_SUBSPACE_KEY_FORMAT_VERSION - 1)
+                    .setContext(context)
+                    .create();
+            assertEquals(FDBRecordStore.SAVE_INDEX_STATE_WITH_SUBSPACE_KEY_FORMAT_VERSION - 1, recordStore.getFormatVersion());
+            commit(context);
+        }
+
+        // Add a new index named "index" on rec_no.
+        final Index index = new Index("index", "rec_no");
+        index.setSubspaceKey(1);
+        builder.addIndex("MySimpleRecord", index);
```

Comment:
Oh, I see. I think it might be clearer if this lived next to the `RecordMetaDataBuilder` definition. It was confusing to me why changing this here would change the meta-data due to the way the meta-data builder was included as a member in the store builder.

Reply:
done